_distutils_hack/__init__.py

```diff
@@ -208,15 +208,20 @@ def __enter__(self):
         insert_shim()
 
     def __exit__(self, exc, value, tb):
-        remove_shim()
+        _remove_shim()
 
 
 def insert_shim():
     sys.meta_path.insert(0, DISTUTILS_FINDER)
 
 
-def remove_shim():
+def _remove_shim():
```

Comment:
This whole module is private, so it's probably not necessary to mark it private, but also no problem if it is.

Reply:
Thank you Jason. The renaming of the function was inspired by this piece of code in `pip`:

```python
try:
    __import__("_distutils_hack").remove_shim()
except (ImportError, AttributeError):
    pass
```

Due to the rename the call would fail with an `AttributeError` that would be simply ignored. Without the renaming, there is some change of the importer being disable on Python >= 3.12.